Redis subagent added

alkk opened the pull request.

alkk committed:
Redis subagent added

2128506 reviewed (commented):
It will fail non-UNICODE build because in few places code assumes that TCHAR is WCHAR.
I'm not sure that character conversion should be to/from system codepage. Usually all such APIs expect UTF-8 encoding - but this should be checked.
Otherwise looks good.

alkk committed:
Fixed Redis subagent character encoding issues

- Fixed non-UNICODE build failures by replacing WCHAR with TCHAR
- Changed character conversions from system codepage to UTF-8
- Replaced WideCharToMultiByteSysLocale/MultiByteToWideCharSysLocale with UTF8 conversion functions
- Fixed swprintf usage to use _sntprintf for TCHAR compatibility

These changes ensure the Redis subagent works correctly with both UNICODE and non-UNICODE builds,
and properly handles UTF-8 encoding from the Redis API.
